packages/lesswrong/server/notificationCallbacks.tsx

```diff
@@ -501,32 +503,44 @@ const sendNewCommentNotifications = async (comment: DbComment) => {
   // if e.g. they're both the author of the post and the author of a comment being replied to)
   let notifiedUsers: Array<string> = [];
 
+
+
   // 1. Notify users who are subscribed to the parent comment
   if (comment.parentCommentId) {
-    const parentComment = await Comments.findOne(comment.parentCommentId)
-    if (parentComment) {
+    const parentComments: {commentId: string, userId: string}[] = await new CommentsRepo().getParentCommentIds({ commentId: comment._id });
+
+    let newReplyUserIds: string[] = [];
+    let newReplyToYouUserIds: string[] = [];
+
+    for (const {commentId, userId} of parentComments) {
       const subscribedUsers = await getSubscribedUsers({
-        documentId: comment.parentCommentId,
+        documentId: commentId,
         collectionName: "Comments",
         type: subscriptionTypes.newReplies,
-        potentiallyDefaultSubscribedUserIds: [parentComment.userId],
+        potentiallyDefaultSubscribedUserIds: [userId],
         userIsDefaultSubscribed: u => u.auto_subscribe_to_my_comments
       })
       const subscribedUserIds = _.map(subscribedUsers, u=>u._id);
-      
+
       // Don't notify the author of their own comment, and filter out the author
       // of the parent-comment to be treated specially (with a newReplyToYou
       // notification instead of a newReply notification).
-      let parentCommentSubscriberIdsToNotify = _.difference(subscribedUserIds, [comment.userId, parentComment.userId])
-      await createNotifications({userIds: parentCommentSubscriberIdsToNotify, notificationType: 'newReply', documentType: 'comment', documentId: comment._id});
-
-      // Separately notify author of comment with different notification, if
-      // they are subscribed, and are NOT the author of the comment
-      if (subscribedUserIds.includes(parentComment.userId) && parentComment.userId !== comment.userId) {
-        await createNotifications({userIds: [parentComment.userId], notificationType: 'newReplyToYou', documentType: 'comment', documentId: comment._id});
-        notifiedUsers = [...notifiedUsers, parentComment.userId];
+      newReplyUserIds = [...newReplyUserIds, ..._.difference(subscribedUserIds, [comment.userId, commentId])]
+
+      // Separately notify authors of replies to their own comments
+      if (subscribedUserIds.includes(userId) && userId !== comment.userId) {
+        newReplyToYouUserIds = [...newReplyToYouUserIds, ...subscribedUserIds]
       }
     }
+
+    // Take the difference as a precaution to prevent double-notifying
+    newReplyUserIds = uniq(_.difference(newReplyUserIds, newReplyToYouUserIds));
+    newReplyToYouUserIds = uniq(newReplyToYouUserIds);
+
+    await createNotifications({userIds: newReplyUserIds, notificationType: 'newReply', documentType: 'comment', documentId: comment._id});
+    await createNotifications({userIds: newReplyToYouUserIds, notificationType: 'newReplyToYou', documentType: 'comment', documentId: comment._id});
```

Comment:
Can be a in a `Promise.all` - dunno if this callback is blocking on anything, though

Reply:
Yep, I've changed this 👍 . Although it isn't blocking so not super critical